inc/admin.php

```diff
@@ -1059,14 +1060,57 @@ function action_builder_content() {
 			false
 		);
 		$panels_data            = SiteOrigin_Panels_Styles_Admin::single()->sanitize_all( $panels_data );
-
+		
 		// Create a version of the builder data for post content
 		SiteOrigin_Panels_Post_Content_Filters::add_filters();
 		$GLOBALS[ 'SITEORIGIN_PANELS_POST_CONTENT_RENDER' ] = true;
 		echo SiteOrigin_Panels::renderer()->render( intval( $_POST['post_id'] ), false, $panels_data );
 		SiteOrigin_Panels_Post_Content_Filters::remove_filters();
 		unset( $GLOBALS[ 'SITEORIGIN_PANELS_POST_CONTENT_RENDER' ] );
-
+		
+		wp_die();
+	}
+	
+	/**
+	 * Get builder content based on the submitted panels_data.
+	 */
+	function action_builder_content_json() {
```

Comment:
This needs a nonce check just to be safe.

Reply:
Yeah, it definitely does. Well spotted.